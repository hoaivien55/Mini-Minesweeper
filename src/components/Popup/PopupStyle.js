import styled from 'styled-components';


const PopupStyle = styled.div`
width: 100vw;
height: 100vh;
position: fixed;
left: 0;
top: 0;
background: #76c3c34d;
z-index: 100;
`;

const PopupContainerStyle = styled.div`
width: 100%;
height: 100%;
max-width: ${(props) => props.maxWidth || "400px"};
max-height: ${(props) => props.maxHeight || "200px"};
min-width: 200px;
min-height: 200px;
margin: 30px auto;
background: white;
border-radius: 12px;
`;

const PopupBodyStyle = styled.div`
width: 100%;
height: ${(props) => props.height || "70%"};
display: flex;
justify-content: center;
align-items: center;
.paragraph {
    font-size: 25px;
}
`;

const PopupFooter = styled.div`
width: 100%;
height: ${(props) => props.height || "30%"};
display: flex;
align-items: center;
justify-content: center;
border-top: 1px solid #80808030;
button:first-child {
    margin-right: 15px;
}
`;

export {
    PopupStyle,
    PopupContainerStyle,
    PopupBodyStyle,
    PopupFooter
}