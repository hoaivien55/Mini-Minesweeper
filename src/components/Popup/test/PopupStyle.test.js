import renderer from 'react-test-renderer';
import { PopupBodyStyle, PopupContainerStyle, PopupFooter, PopupStyle } from '../PopupStyle';


it('PopupContainerStyle: Virtual DOM snapshot', () => {
    const tree = renderer
      .create(<PopupContainerStyle />)
      .toJSON();
    expect(tree).toMatchSnapshot();
});

it('PopupContainerStyle: Virtual DOM snapshot change max-width and max-height', () => {
    const tree = renderer
      .create(<PopupContainerStyle maxWidth={"1px"} maxHeight={"1px"} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
});


it('PopupStyle: Virtual DOM snapshot', () => {
    const tree = renderer
      .create(<PopupStyle />)
      .toJSON();
    expect(tree).toMatchSnapshot();
});

it('PopupBodyStyle: Virtual DOM snapshot', () => {
    const tree = renderer
      .create(<PopupBodyStyle />)
      .toJSON();
    expect(tree).toMatchSnapshot();
});

it('PopupFooter: Virtual DOM snapshot', () => {
    const tree = renderer
      .create(<PopupFooter />)
      .toJSON();
    expect(tree).toMatchSnapshot();
});