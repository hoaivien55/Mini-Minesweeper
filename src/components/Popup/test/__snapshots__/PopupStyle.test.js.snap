// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopupBodyStyle: Virtual DOM snapshot 1`] = `
.c0 {
  width: 100%;
  height: 70%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .paragraph {
  font-size: 25px;
}

<div
  className="c0"
/>
`;

exports[`PopupContainerStyle: Virtual DOM snapshot 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  max-width: 400px;
  max-height: 200px;
  min-width: 200px;
  min-height: 200px;
  margin: 30px auto;
  background: white;
  border-radius: 12px;
}

<div
  className="c0"
/>
`;

exports[`PopupContainerStyle: Virtual DOM snapshot change max-width and max-height 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  max-width: 1px;
  max-height: 1px;
  min-width: 200px;
  min-height: 200px;
  margin: 30px auto;
  background: white;
  border-radius: 12px;
}

<div
  className="c0"
/>
`;

exports[`PopupFooter: Virtual DOM snapshot 1`] = `
.c0 {
  width: 100%;
  height: 30%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-top: 1px solid #80808030;
}

.c0 button:first-child {
  margin-right: 15px;
}

<div
  className="c0"
/>
`;

exports[`PopupStyle: Virtual DOM snapshot 1`] = `
.c0 {
  width: 100vw;
  height: 100vh;
  position: fixed;
  left: 0;
  top: 0;
  background: #76c3c34d;
  z-index: 100;
}

<div
  className="c0"
/>
`;
